src/apiClient.ts

```diff
@@ -1,11 +1,10 @@
-export interface ApiClientConfig {
-  carServiceUrl?: string
-  searchServiceUrl?: string
-  catalogueServiceUrl?: string
-  dealerServiceUrl?: string
-  optionServiceUrl?: string
-  analyticsServiceUrl?: string
-  userNotificationServiceUrl?: string
+export interface ApiClientConfiguration {
+  host: string
+  debug?: boolean
+}
+
+interface ApiClientConfig {
```

Comment:
Why do you need an additional internal type you you default the configuration to an empty object in the constructor?

Reply:
Because `host` is required in the external type and optional in the internal one.
As in you can't configure the `ApiClient` without the `host` you'll get the type error

Reply:
Oh nvm messed that up 😂